### PR TITLE
Added NuGet.config to TemplateFiles

### PR DESCRIPTION
**Pull request by CDiaz-MS**
## Overview
While retrieving the MRTK NuGet packages via MSBuildForUnity, you first need to add ```https://pkgs.dev.azure.com/UnityDeveloperTools/MSBuildForUnity/_packaging/UnityDeveloperTools/nuget/v3/index.json ``` as a source.  This change removes adding an additional source to enable MRTK NuGet package installation with one step.

## Changes
- Added NuGet.config to TemplateFiles.cs
- Moved Nuget.config to the TemplateFiles directory

## Fixes
- #99

### CODE REVIEW

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/NuGet.config [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
```
Comment (andreiborodin) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/NuGet.config]: I don't think this line is needed, as this logic simply "adds" a source to existing set of sources, which should have NuGet.org by default.

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs [hunk 2]
```diff
@@ -382,7 +382,7 @@ private static void RegenerateEverything(UnityProjectInfo unityProjectInfo, bool
 
                 foreach (string otherFile in TemplateFiles.Instance.OtherFiles)
                 {
-                    File.Copy(otherFile, Path.Combine(Utilities.MSBuildProjectFolder, Path.GetFileName(otherFile)));
+                    File.Copy(otherFile, Path.Combine(Utilities.AssetPath, Path.GetFileName(otherFile)));
```
Comment (andreiborodin) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs]: I think you may want to undo this, as we want to copy "otherfiles" from templates to the MSBuildProjectFolder.

What you want to do, is above the ForEach, copy the NuGet.config template to Utilities.AssetPath

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Templates/TemplateFiles.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public class TemplateFiles
         private const string SpecifcPlatformPropsTemplateRegex = @"[a-zA-Z]+\.[a-zA-Z]+\.[a-zA-Z0-9]*\.props.template";
         private const string PluginMetaFileTemplateRegex = @"Plugin\.([a-zA-Z]*)\.meta.template";
         private const string BuildProjectsTemplateName = "BuildProjects.proj.template";
+        private const string NugetConfigFileName = "NuGet.config";
```
Comment (andreiborodin) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Templates/TemplateFiles.cs]: You also want to add the Property for NuGetConfigFile, and set the property as done below by something like "BuildProjectsTemplateName"

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/NuGet.config (hunk 1 above)
Comment (CDiaz-MS) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/MSBuildTemplates/NuGet.config]: Removed

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Templates/TemplateFiles.cs (hunk 3 above)
Comment (CDiaz-MS) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Templates/TemplateFiles.cs]: Fixed

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs (hunk 2 above)
Comment (CDiaz-MS) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs]: Fixed

**Pull request merged**